src/operators/InterestDistributor.sol

```diff
@@ -0,0 +1,104 @@
+        uint128 prevShares = user.shares;
+
+        (address asset,) = poolManager.getVaultAsset(vault);
+        (uint128 currentPrice, uint64 priceLastUpdated) =
+            poolManager.getTranchePrice(vault_.poolId(), vault_.trancheId(), asset);
+        uint128 currentShares = IERC20(vault_.share()).balanceOf(controller).toUint128();
+
+        // Calculate request before updating user.shares, so it is based on the balance at the last price update.
+        // Assuming price updates coincide with epoch fulfillments, this results in only requesting
+        // interest on the previous outstanding balance before the new fulfillment.
+        uint128 request = priceLastUpdated > user.lastUpdate
+            ? _computeRequest(prevShares, currentShares, user.peak, uint96(currentPrice))
+            : 0;
+
+        user.lastUpdate = uint32(priceLastUpdated);
+        if (currentPrice > user.peak) user.peak = uint96(currentPrice);
+        user.shares = currentShares - request;
+        _users[vault][controller] = user;
+
+        if (request > 0) {
+            vault_.requestRedeem(request, controller, controller);
+            emit InterestRedeemRequest(vault, controller, user.peak, uint96(currentPrice), request);
+        }
+
+        if (user.shares != prevShares) {
+            emit OutstandingSharesUpdate(vault, controller, prevShares, user.shares);
+        }
+    }
+
+    /// @inheritdoc IInterestDistributor
+    function clear(address vault, address controller) external {
+        require(!IERC7540Vault(vault).isOperator(controller, address(this)), "InterestDistributor/still-an-operator");
+        require(_users[vault][controller].lastUpdate > 0, "InterestDistributor/unknown-controller");
+
+        delete _users[vault][controller];
+        emit Clear(vault, controller);
+    }
+
+    /// @inheritdoc IInterestDistributor
+    function pending(address vault, address controller) external view returns (uint128 shares) {
+        InterestDetails memory user = _users[vault][controller];
+        IERC7540Vault vault_ = IERC7540Vault(vault);
+        (uint128 currentPrice, uint64 priceLastUpdated) =
+            poolManager.getTranchePrice(vault_.poolId(), vault_.trancheId(), vault_.asset());
+        if (user.lastUpdate == uint32(priceLastUpdated)) return 0;
+        shares =
+            _computeRequest(user.shares, IERC20(vault_.share()).balanceOf(controller), user.peak, uint96(currentPrice));
+    }
+
+    /// @dev Calculate shares to redeem based on outstandingShares * ((currentPrice - prevPrice) / currentPrice)
+    function _computeRequest(uint128 outstandingShares, uint256 currentShares, uint96 prevPrice, uint96 currentPrice)
+        internal
+        pure
+        returns (uint128)
+    {
+        if (outstandingShares == 0 || currentPrice <= prevPrice) return 0;
+
+        return MathLib.min(
+            uint256(outstandingShares).mulDiv(currentPrice - prevPrice, currentPrice, MathLib.Rounding.Down),
```

Comment:
Isn't this supposed to be divided by the price decimals? I assumed 10**18 but it should be calculated.
```suggestion
            uint256(outstandingShares).mulDiv(currentPrice - prevPrice, 10**18, MathLib.Rounding.Down),
```